File: JAVA/015_Lattice_Paths.java
class TestMain
{
	public static void main(String[] args)
	{
		System.out.println(noOfPaths(20l));
	}

	private static double noOfPaths(long side)
	{
		long upper = 2 * side;
		double result=1.0/factorial(side);
		long counter = side;
		for(;counter>0l;counter--)
		{
			result*=(upper--);
			System.out.println("r "+result);
		}
		
		return result;
	}
	
	private static long factorial(long n)
	{
		if(n>1)
			return n * factorial(n-1);
		
			return 1l;
		
	}
}

File: CPP/015_Lattice_Paths.cpp
#include <iostream>

using namespace std;
long long factorial(long long n)
        {
                if(n>1ll)
                        return n * factorial(n-1);
                
                        return 1ll;
                
        }
        long long noOfPaths(long long side)
        {
                long long upper = 2ll * side;
                long double result=1.0/factorial(side);
                long long counter = side;
                for(;counter>0ll;counter--)
                {
                        result*=(upper--);
                }
                
                return (long long)result;
        }
        int main()
        {
            cout << noOfPaths(10ll) << endl;
            return 0;
        }